docs/api/implementation-guide.md

```diff
@@ -0,0 +1,67 @@
+---
+title: NuGet Server Implementation Guide
+description: Guidelines and recommendations to anyone implementing the NuGet Server API in their own package repository
+author: zivkan
+ms.author: zivkan
+ms.date: 07/29/2023
+ms.topic: conceptual
+---
+
+# NuGet Server Implementation Guide
+
+Prior to 2019, NuGet package repositories could copy the package metadata from the `nupkg`'s [`nuspec` file](../reference/nuspec.md), and return the results unmodified in [search](./search-query-service-resource.md) and [package metadata](./registration-base-url-resource.md) requests.
+Since then, NuGet has evolved to provide developers with a richer experience, and this requires package repositories to do additional work in order to provide the additional value to developers.
+
+## URL structure for private feeds
+
+As described in [the overview of the NuGet API](./overview.md), the starting URL for all NuGet server communication is [the service index](./service-index.md).
+This document contains the URLs for all other resources that NuGet clients will query.
+As of NuGet 6.7 (Visual Studio & MSBuild 17.7, and .NET SDK 7.0.400), NuGet uses [.NET's `HttpClientHandler.PreAuthenticate](/dotnet/api/system.net.http.httpclienthandler.preauthenticate), which only works when subsequent URLs are in the same virtual directory, or a subdirectory, of a URL that has previously been authenticated.
+This will dramatically reduce the number of unauthenticated HTTP requests sent to the server, and therefore will reduce your server workload.
+
+For example, imagine the service index is located at https://pkgs.contoso.com/nuget/v3/service/index.json, but the search resource is at https://pkgs.contoso.com/nuget/v3/search.
+Since the search URL is not in the same directory as the service index, or a subdirectory, `HttpClientHandler`'s credential cache won't use the credential on the first request, forcing the server to respond with an HTTP 401 response, and then `HttpClientHandler` will re-request the URL with credentials.
+If the search resource was instead at https://pkgs.contoso.com/nuget/v3/service/search, the first request to the search resource will be authenticated since the search endpoint is in the same directory as the service index.
+Similarly, a `RegistrationBaseUrl` of https://pkgs.contoso.com/nuget/v3/registration/ will always try an unauthenticated request first, since the URL is not a subdirectory of the service index's location, but https://pkgs.contoso.com/nuget/v3/service/registration/ will be pre-authenticated since it's a subdirectory of the service index's location.
```

Comment:
This is very wordy, and I'm not sure it's clear.  What do you all think if I replace it with this table?

|URL|Will PreAuthenticate?|
|--|--|
|https://pkgs.contoso.com/nuget/v3/service/index.json|n/a, this is the service index|
|https://pkgs.contoso.com/nuget/v3/search|no, not in the same directory, or subdirectory, of the service index.|
|https://pkgs.contoso.com/nuget/v3/registration/|no, not in the same directory, or subdirectory, of the service index.|
|https://pkgs.contoso.com/nuget/v3/service/search|Yes, in the same directory as the service index.|
|https://pkgs.contoso.com/nuget/v3/service/registration|yes, in a subdirectory of the service index.|
|https://pkgs.contoso.com/nuget/v3/{guid}/search|Only if the service index URL redirected to https://pkgs.contoso.com/nuget/v3/{guid}/index.json. Otherwise, even if the GUID is the canonical name of the feed and all other names are aliases, since it's not the URL that `HttpClientHandler` used in its credential cache, these requests will not be pre-authenticated.|

Reply:
I like the able better.

Reply:
FWIW, I found the text perfectly readable but a table does add visual variety. Both seem like good options :)

Reply:
Why not put the resource name instead of the URL in the first column? (and link the resource name to the respective doc)

Reply:
@joelverhagen I don't understand the suggestion. Can you show an example?

I'm not sure if this answers your question, but one of the issues is that github limits the width it renders markdown, so if there are too many columns, or if there's too much text, everything gets wrapped. For URLs in particular, this gets harder to read.